### ISSUE TITLE
`UsedAllocations` returns an error when the account has no used allocation yet

### ISSUE THREAD

**Issue by lubtd** [tendermint/spn#639]
The query returns a Not Found error if an account has not yet used allocations for participation
```
spnd q participation show-used-allocations spn1ezptsm3npn54qx9vvpah4nymre59ykr9exx2ul
Error: rpc error: code = NotFound desc = rpc error: code = NotFound desc = not found: key not found
```

It can be argued that this error is logic since the account has no entry in the store yet for this value. However, https://github.com/tendermint/spn/pull/630 bring `TotalAllocations` and `AvailailableAllocations` that return values as long as the account exists
```
spnd q participation show-total-allocations spn1aqn8ynvr3jmq67879qulzrwhchq5dtrvtx0nhe
totalAllocations: "0"
spnd q participation show-total-allocations spn1ezptsm3npn54qx9vvpah4nymre59ykr9exx2ul
totalAllocations: "100000"
spnd q participation show-available-allocations spn1ezptsm3npn54qx9vvpah4nymre59ykr9exx2ul
availableAllocations: "100000"
```

In my opinion, we should return `0` for `UsedAllocations` if the account doesn't use allocations so we have consistent values with `TotalAllocations`  and `AvailailableAllocations`

**Comment by giunatale** [comment]
Note: I think it returns 0 as long as the account is valid, no checks for existence as far as I can see. Can you confirm?

**Comment by giunatale** [comment]
Also, I'd rather do this only in the query, and leave the keeper store getter as it is. WDYT?

**Comment by lubtd** [comment]
Yes, the idea is that the query filter the non-existence and return 0 un this case.
Currently even if the account is valid, the query return an error if the account never participated in an auction
